The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8217016165/job/22472227017) [(plain)](https://github.com/rust-lang/rust/commit/dbe2636dd2774c2d37ef42d73e8f0d3a1b1355bd/checks/22472227017/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=Nadrieril
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_4232da13-3fd4-4155-b9fb-f14d847de6f5
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=fc195cd73da4e092584e9a80aa6d342415bb98e3
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_4232da13-3fd4-4155-b9fb-f14d847de6f5
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_4232da13-3fd4-4155-b9fb-f14d847de6f5
GITHUB_TRIGGERING_ACTOR=Nadrieril
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/122255/merge
GITHUB_WORKFLOW_SHA=fc195cd73da4e092584e9a80aa6d342415bb98e3
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_20_X64=/opt/hostedtoolcache/go/1.20.14/x64
---
#12 writing image sha256:ade963e0370c46f87b4ebcba8321b4a4ded5091d1f524b99993f0c86f8c3d12d done
#12 naming to docker.io/library/rust-ci done
#12 DONE 10.2s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Sat Mar  9 20:29:55 UTC 2024
  network time: Sat, 09 Mar 2024 20:29:55 GMT
  network time: Sat, 09 Mar 2024 20:29:55 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'build.optimized-compiler-builtins', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
   Compiling std_detect v0.1.5 (/checkout/library/stdarch/crates/std_detect)
   Compiling miniz_oxide v0.7.2
   Compiling addr2line v0.21.0
   Compiling std v0.0.0 (/checkout/library/std)
error[E0004]: non-exhaustive patterns: `TryLockError::Poisoned(_)` not covered
    |
270 |         match *self {
270 |         match *self {
    |               ^^^^^ pattern `TryLockError::Poisoned(_)` not covered
note: `TryLockError<T>` defined here
   --> library/std/src/sync/poison.rs:137:10
    |
    |
137 | pub enum TryLockError<T> {
...
...
141 |     Poisoned(#[stable(feature = "rust1", since = "1.0.0")] PoisonError<T>),
    = note: the matched value is of type `TryLockError<T>`
    = note: the matched value is of type `TryLockError<T>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
273 ~             TryLockError::WouldBlock => "WouldBlock".fmt(f),
274 ~             TryLockError::Poisoned(_) => todo!(),


error[E0004]: non-exhaustive patterns: `TryLockError::Poisoned(_)` not covered
    |
281 |         match *self {
281 |         match *self {
    |               ^^^^^ pattern `TryLockError::Poisoned(_)` not covered
note: `TryLockError<T>` defined here
   --> library/std/src/sync/poison.rs:137:10
    |
    |
137 | pub enum TryLockError<T> {
...
...
141 |     Poisoned(#[stable(feature = "rust1", since = "1.0.0")] PoisonError<T>),
    = note: the matched value is of type `TryLockError<T>`
    = note: the matched value is of type `TryLockError<T>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
284 ~             TryLockError::WouldBlock => "try_lock failed because the operation would block",
285 ~             TryLockError::Poisoned(_) => todo!(),


error[E0004]: non-exhaustive patterns: `TryLockError::Poisoned(_)` not covered
    |
294 |         match *self {
294 |         match *self {
    |               ^^^^^ pattern `TryLockError::Poisoned(_)` not covered
note: `TryLockError<T>` defined here
   --> library/std/src/sync/poison.rs:137:10
    |
    |
137 | pub enum TryLockError<T> {
...
...
141 |     Poisoned(#[stable(feature = "rust1", since = "1.0.0")] PoisonError<T>),
    = note: the matched value is of type `TryLockError<T>`
    = note: the matched value is of type `TryLockError<T>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
297 ~             TryLockError::WouldBlock => "try_lock failed because the operation would block",
298 ~             TryLockError::Poisoned(_) => todo!(),

For more information about this error, try `rustc --explain E0004`.
error: could not compile `std` (lib) due to 3 previous errors
Build completed unsuccessfully in 0:14:48
```

</details>

--- a/library/core/src/lib.rs
+++ b/library/core/src/lib.rs
@@ -203,8 +203,10 @@
 // Language features:
 // tidy-alphabetical-start
 #![cfg_attr(bootstrap, feature(diagnostic_namespace))]
+#![cfg_attr(bootstrap, feature(exhaustive_patterns))]
 #![cfg_attr(bootstrap, feature(platform_intrinsics))]
 #![cfg_attr(not(bootstrap), feature(freeze_impls))]
+#![cfg_attr(not(bootstrap), feature(min_exhaustive_patterns))]
 #![feature(abi_unadjusted)]
 #![feature(adt_const_params)]
 #![feature(allow_internal_unsafe)]
@@ -229,7 +231,6 @@
 #![feature(doc_cfg_hide)]
 #![feature(doc_notable_trait)]
 #![feature(effects)]
-#![feature(exhaustive_patterns)]
 #![feature(extern_types)]
 #![feature(fundamental)]
 #![feature(generic_arg_infer)]

--- a/library/std/src/lib.rs
+++ b/library/std/src/lib.rs
@@ -270,7 +270,9 @@
 //
 // Language features:
 // tidy-alphabetical-start
+#![cfg_attr(bootstrap, feature(exhaustive_patterns))]
 #![cfg_attr(bootstrap, feature(platform_intrinsics))]
+#![cfg_attr(not(bootstrap), feature(min_exhaustive_patterns))]
 #![feature(alloc_error_handler)]
 #![feature(allocator_internals)]
 #![feature(allow_internal_unsafe)]
@@ -289,7 +291,6 @@
 #![feature(doc_masked)]
 #![feature(doc_notable_trait)]
 #![feature(dropck_eyepatch)]
-#![feature(exhaustive_patterns)]
 #![feature(if_let_guard)]
 #![feature(intra_doc_pointers)]
 #![feature(lang_items)]

--- a/library/std/src/sync/poison.rs
+++ b/library/std/src/sync/poison.rs
@@ -270,6 +270,8 @@ impl<T> fmt::Debug for TryLockError<T> {
         match *self {
             #[cfg(panic = "unwind")]
             TryLockError::Poisoned(..) => "Poisoned(..)".fmt(f),
+            #[cfg(not(panic = "unwind"))]
+            TryLockError::Poisoned(ref p) => match p._never {},
             TryLockError::WouldBlock => "WouldBlock".fmt(f),
         }
     }
@@ -281,6 +283,8 @@ impl<T> fmt::Display for TryLockError<T> {
         match *self {
             #[cfg(panic = "unwind")]
             TryLockError::Poisoned(..) => "poisoned lock: another task failed inside",
+            #[cfg(not(panic = "unwind"))]
+            TryLockError::Poisoned(ref p) => match p._never {},
             TryLockError::WouldBlock => "try_lock failed because the operation would block",
         }
         .fmt(f)
@@ -294,6 +298,8 @@ impl<T> Error for TryLockError<T> {
         match *self {
             #[cfg(panic = "unwind")]
             TryLockError::Poisoned(ref p) => p.description(),
+            #[cfg(not(panic = "unwind"))]
+            TryLockError::Poisoned(ref p) => match p._never {},
             TryLockError::WouldBlock => "try_lock failed because the operation would block",
         }
     }
@@ -303,6 +309,8 @@ impl<T> Error for TryLockError<T> {
         match *self {
             #[cfg(panic = "unwind")]
             TryLockError::Poisoned(ref p) => Some(p),
+            #[cfg(not(panic = "unwind"))]
+            TryLockError::Poisoned(ref p) => match p._never {},
             _ => None,
         }
     }

